ietf/group/utils.py

```diff
@@ -157,13 +158,19 @@ def can_manage_session_materials(user, group, session):
 def can_manage_some_groups(user):
     if not user.is_authenticated:
         return False
+    authroles = set(list(chain.from_iterable(GroupFeatures.objects.values_list("groupman_authroles",flat=True))))
+    extra_role_qs = dict()
     for gf in GroupFeatures.objects.all():
-        for authrole in gf.groupman_authroles:
-            if has_role(user, authrole):
-                return True
-            if Role.objects.filter(name__in=gf.groupman_roles, group__type_id=gf.type_id, person__user=user).exists():
-                return True
-    return False          
+        extra_role_qs[f"{gf.type_id} groupman roles"] = Q(name__in=gf.groupman_roles, group__type_id=gf.type_id, group__state__in=["active","bof", "proposed"])
+    return has_role(user, authroles, extra_role_qs = extra_role_qs)
+
+    # for gf in GroupFeatures.objects.all():
+    #     for authrole in gf.groupman_authroles:
+    #         if has_role(user, authrole):
+    #             return True
+    #         if Role.objects.filter(name__in=gf.groupman_roles, group__type_id=gf.type_id, person__user=user).exists():
+    #             return True
+    # return False          
```

Comment:
Go ahead and remove the commented out code? (or annotate why it's here if you want to keep it to document what the algorithm is doing)